response.go

```diff
@@ -262,6 +262,52 @@ func (r *Response) StatusRange(rn StatusRange) *Response {
 	return r
 }
 
+// StatusList succeeds if response matches with any given status code list
+//
+// Example:
+//
+//	resp := NewResponse(t, response)
+//	resp.StatusList(http.StatusForbidden, http.StatusUnauthorized)
+func (r *Response) StatusList(values ...int) *Response {
+	r.chain.enter("StatusList()")
+	defer r.chain.leave()
+
+	if r.chain.failed() {
+		return r
+	}
+
+	if len(values) == 0 {
+		r.chain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected empty status list"),
+			},
+		})
+		return r
+	}
+
+	var found bool
+	for _, v := range values {
+		if v == r.httpResp.StatusCode {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		r.chain.fail(AssertionFailure{
+			Type:     AssertBelongs,
+			Actual:   &AssertionValue{statusCodeText(r.httpResp.StatusCode)},
+			Expected: &AssertionValue{AssertionList{statusListText(values)}},
```

Comment:
`AssertionList` is `[]interface{}`.

Thus, `AssertionList{statusListText(values)}` means:

```
    []interface{
        statusListText(values),
    }
```

i.e. it is a list of single element which itself is a list of strings.

But we instead want just a list of strings, of type AssertionList. To achieve this, we can modify statusListText to return `[]interface{}` instead of `[]string` and then replace `AssertionList{statusListText(values)}` with `AssertionList(statusListText(values))`.

Reply:
I added validation to catch this specific kind of bug: 532fdc1f39aa92436c97f156e36b3b191f05d722

Reply:
ah yes, sorry missed that. Code updated, hopefully it's correct now.
Just one note from me, since func `statusListText()` now return `[]interface{}` it won't as reusable as before, but I think it's ok as it's very specific helper function. Feel free if you have more comments. Thanks!

Reply:
Yep, I think that's not a big deal